### PR TITLE
Add handling of game start

### LINKED ISSUE #10: Start game
Create a simple UI that will allow the game creator to start a game when there are 2-8 players in the room.

### PR DESCRIPTION
**Pull request by mikolaj-pirog**
Closes #10

### CODE REVIEW

File: app/src/main/java/com/pokerio/app/MainActivity.kt [hunk 1]
```diff
@@ -63,6 +65,13 @@ fun MainActivityComposable() {
             navController.popBackStack("home", inclusive = false)
         }
     }
+    GameState.onGameStart = {
+        PokerioLogger.debug("Called onGameStart")
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/MainActivity.kt]: Useless debug

Comment (Kwasow) [inline app/src/main/java/com/pokerio/app/MainActivity.kt]: This debug is not necessary anymore

**Review by Kwasow — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 2]
```diff
@@ -26,23 +26,27 @@ object GameState {
...
     var onGameReset = {}
+    var onGameStart = {}
     private val playerJoinedCallbacks = HashMap<Int, (Player) -> Unit>()
     private val playerRemovedCallbacks = HashMap<Int, (Player) -> Unit>()
     private var settingsChangedCallback = HashMap<Int, () -> Unit>()
     private var nextId = 0
 
     // Constants
-    private const val BASE_URL = "http://158.101.160.143:42069"
+    private const val BASE_URL = "http://10.0.2.2:42069"
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: Wrong url

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 3]
```diff
@@ -251,14 +256,39 @@ object GameState {
...
+            try {
+                val myID = FirebaseMessaging.getInstance().token.await()
+
+                // Prepare url
+                val urlString = "/startGame?creatorToken=$myID"
+                val url = URL(baseUrl + urlString)
+
+                url.readText()
+
+                // onGameStart()
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: No onSuccess() call

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 4]
```diff
@@ -334,7 +364,17 @@ object GameState {
         }
     }
 
-    private fun sha256(string: String): String {
+    fun startGame(data: Map<String, String>) {
+        card1 = Card(data["card1"]!!.slice(0..1), data["card1"]!!.slice(2..2))
+        card2 = Card(data["card2"]!!.slice(0..1), data["card2"]!!.slice(2..2))
+        val gameInfoJsonString = data["startedGameInfo"]!!
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: It is unused

**Review by Kwasow — APPROVED** [review]

**Review by sormys — APPROVED** [review]

**Pull request merged**